nwb_conversion_tools/utils/neo.py

```diff
@@ -0,0 +1,570 @@
+    accepted by pynwb.icephys.IntracellularElectrode:
+    - name – the name of this electrode
+    - device – the device that was used to record from this electrode
+    - description – Recording description, description of electrode (e.g., whole-cell, sharp, etc) COMMENT: Free-form text (can be from Methods)
+    - slice – Information about slice used for recording.
+    - seal – Information about seal used for recording.
+    - location – Area, layer, comments on estimation, stereotaxis coordinates (if in vivo, etc).
+    - resistance – Electrode resistance COMMENT: unit: Ohm.
+    - filtering – Electrode specific filtering.
+    - initial_access_resistance – Initial access resistance.
+
+    Args:
+        neo_reader ([type]): Neo reader
+        electrodes_ids (list): List of electrodes ids.
+        block (int, optional): Block id. Defaults to 0.
+
+    Returns:
+        list: List of dictionaries containing electrodes metadata
+    """
+    return []
+
+
+# Get number of electrodes
+def get_number_of_electrodes(neo_reader) -> int:
+    """
+    Get number of electrodes from Neo reader
+
+    Args:
+        neo_reader ([type]): Neo reader
+
+    Returns:
+        int: number of electrodes
+    """
+    return len(neo_reader.header["signal_channels"])
+
+
+# Get number of segments
+def get_number_of_segments(neo_reader, block: int = 0) -> int:
+    """
+    Get number of segments from Neo reader
+
+    Args:
+        neo_reader ([type]): Neo reader
+        block (int, optional): Block id. Defaults to 0.
+
+    Returns:
+        int: number of electrodes
+    """
+    return neo_reader.header["nb_segment"][block]
+
+
+# Get command traces (e.g. voltage clamp command traces)
+def get_command_traces(neo_reader, block: int = 0, segment: int = 0, cmd_channel: int = 0) -> Tuple[list, str, str]:
+    """
+    Get command traces (e.g. voltage clamp command traces).
+
+    Args:
+        neo_reader ([type]): [description]
+        block (int, optional): [description]. Defaults to 0.
+        segment (int, optional): [description]. Defaults to 0.
+        cmd_channel (int, optional): ABF command channel (0 to 7). Defaults to 0.
+
+    Returns:
+        list: [description]
+    """
+    traces, titles, units = neo_reader.read_raw_protocol()
+    return traces[segment][cmd_channel], titles[segment][cmd_channel], units[segment][cmd_channel]
+
+
+# Get gain (to Volt or Ampere) from unit in string format
+def get_gain_from_unit(unit: str) -> float:
+    """
+    Get gain (to Volt or Ampere) from unit in string format.
+
+    Args:
+        unit (str): Unit as string. E.g. pA, mV, uV, etc...
+
+    Returns:
+        float: gain to Ampere or Volt
+    """
+    if unit in ["pA", "pV"]:
+        gain = 10 ** -12
+    elif unit in ["nA", "nV"]:
+        gain = 10 ** -9
+    elif unit in ["uA", "uV"]:
+        gain = 10 ** -6
+    elif unit in ["mA", "mV"]:
+        gain = 10 ** -3
+    elif unit in ["A", "V"]:
+        gain = 10 ** 0
```

Comment:
You should handle the case outside this list. It happen for some files.

Reply:
Im now passing a warning message, anything else you suggest, @samuelgarcia?